File: source-base/sep.backend.v1/Data/Entities/BusRoute.cs
namespace sep.backend.v1.Data.Entities
{
    public class BusRoute : BaseEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public int SchoolId { get; set; }
        public int Status { get; set; }
        public virtual School? School { get; set; }
        public virtual ICollection<BusStop>? BusStops { get; set; }
        public virtual ICollection<Bus>? Buses { get; set; }


    }
}

File: source-base/source.base.fe/src/types/model/bus-route.ts
export interface BusRoute {
  id: number;
  name: string;
  description: string;
  status: number;
}

export interface CreateBusRoute {
  name?: string;
  description?: string;
  status?: number;
}
export interface ErrorResponseCreateBusRoute {
  Name?: string[];
  Description?: string[];
  Status?: number[];
  }

export interface ResponseBusRouteData {
  pageNumber: number;
  pageSize: number;
  firstPage: string;
  lastPage: string;
  totalPages: number;
  totalRecords: number;
  nextPage: string | null;
  previousPage: string | null;
  data: BusRoute[];
}

export interface RequestBusRouteIndex {
  pageNumber: number | null;
  pageSize: number | null;
  status?: number | null;
  keyword?: string | null;
}
export interface BusStop {
  id: number;
  name: string;
  pickUpTime: string;
  returnTime: string;
  address: string;
  busRouteId: number;
  status: number;
}

export interface Bus {
  id: number;
  name: string;
  driverName: string;
  driverPhone: string;
  licensePlate: string;
  seatNumber: number;
  busRouteId: number;
  status: number;
}

export interface BusRouteDetail {
  id: number;
  name: string;
  description: string;
  status: number;
  busStops: BusStop[] | null;
  buses: Bus[] | null;
}